core/src/raw/oio/read/mod.rs

```diff
@@ -51,3 +51,4 @@ pub use lazy_read::LazyReader;
 
 mod buffer_reader;
 pub use buffer_reader::BufferReader;
+pub use buffer_reader::MaybeBufferReader;
```

Comment:
I dislike the idea of exposing another non-composable API like this. How about adding some thing like [opendal::raw::oio::TwoWaysWriter](https://opendal.apache.org/docs/rust/opendal/raw/oio/enum.TwoWaysWriter.html)? We can extand `TwoWaysWriter` / `ThreeWaysWriter` into `TwoWay` / `ThreeWay`.

We can finish it in a new PR and update it here.

Reply:
And maybe, we can add `FourWay` to replace `CompleteReader` too?

Reply:
Cool, I will refactor it.

Reply:
> Cool, I will refactor it.

Thanks a lot!